motion.c

```diff
@@ -3214,6 +3214,62 @@ int myfclose(FILE* fh)
 }
 
 /**
+ * mystrftime_long
+ *
+ *   Motion-specific long form of format specifiers.
+ *
+ * Parameters:
+ *
+ *   cnt        - current thread's context structure.
+ *   width      - width associated with the format specifier.
+ *   word       - beginning of the format specifier's word.
+ *   l          - length of the format specifier's word.
+ *   out        - output buffer where to store the result. Size: PATH_MAX.
+ *
+ * This is called if a format specifier with the format below was found:
+ *
+ *   % { word }
+ *
+ * As a special edge case, an incomplete format at the end of the string
+ * is processed as well:
+ *
+ *   % { word \0
+ *
+ * Any valid format specified width is supported, e.g. "%12{host}".
+ *
+ * The following specifier keywords are currently supported:
+ *
+ * host    Replaced with the name of the local machine (see gethostname(2)).
+ * fps     Equivalent to %fps.
+ */
+static void mystrftime_long (const struct context *cnt,
+                             int width, const char *word, int l, char *out)
+{
+#define SPECIFIERWORD(k) ((strlen(k)==l) && (!strncmp (k, word, l)))
+
+    if (SPECIFIERWORD("host")) {
+        char host[PATH_MAX];
+        gethostname (host, PATH_MAX);
```

Comment:
I wonder if we should call gethostname only once at startup and save it as a string. So we don't need to call this function every single time. @jogu @Mr-Dave ?